docs/source/upgrading_your_components.md

```diff
@@ -0,0 +1,328 @@
+of making errors.
+
+We will talk more about these configuration files in the relevant steps.
+
+### Ledger backup and restore
+
+While we will demonstrate the process for backing up ledger data in this tutorial,
+it is not strictly required to backup the ledger data of a peer or an ordering
+node (assuming the node is part of a larger group of nodes in an ordering service).
+This is because, even in the worst case of catastrophic failure of a peer (such
+as a disk failure), the peer can be brought up with no ledger at all. You can
+then have the peer re-join the desired channels and as a result, the peer will
+automatically create a ledger for each of the channels and will start receiving
+the blocks via regular block transfer mechanism from either the ordering service
+or the other peers in the channel. As the peer processes blocks, it will also build
+up its state database.
+
+Still, some organizations may want to be more self-reliant and want to backup
+their ledger data so as to be able to recover from catastrophic failures and
+resume block processing from the point of the last backup. In addition, ledger
+data backups may help to expedite the addition of a new peer, which can be
+achieved by backing up the ledger data from one peer and starting the new peer
+with the backed up ledger data.
+
+This tutorial presumes that the file path to the ledger data has not been changed
+from the default value of `/var/hyperledger/production/<node>`. If this location
+has been changed for your nodes, enter the path to the data on your ledgers in
+the commands below.
+
+Note that there will be data for both the ledger and chaincodes at this file
+location. While it is a best practice to backup both, it is possible to skip
+the `stateLeveldb`, `historyLeveldb`, `chains/index` folders at
+`/var/hyperledger/production/<node>/ledgersData`. While skipping these folders
+reduces the storage needed for the backup, the peer recovery from the backed up
+data may take more time as these ledger artifacts will be re-constructed when the
+peer starts.
+
+If using CouchDB as state database, there will be no `stateLeveldb` directory,
+as the state database data would stored within CouchDB instead. But similarly,
+if peer starts up and finds CouchDB databases are missing or at lower block height
+(based on using an older CouchDB backup), the state database will be automatically
+re-constructed to catch up to current block height. Therefore, if you backup peer
+ledger data and CouchDB data separately, ensure that the CouchDB backup is always
+older than the peer backup.
+
+## Prerequisites
+
+Before attempting to upgrade components, make sure you have all of the dependencies
+on your machine as described in [Prerequisites](./prereqs.html). This will not only ensure that
+you have the latest binaries, but the latest version of all dependencies.
+
+Then make sure you're in the right branch of the Fabric repo:
+
+```
```

Comment:
This seems like maybe it was originally intended for fabric-samples? But nothing I've read would make me understand that.

I rather suspect the git oriented bits should be removed.

Reply:
Done.